### DISCUSSION TITLE
Unable to set up new endpoints

### DISCUSSION THREAD

**Discussion by ylkuo** [mozilla/hubs#3172] [Hubs Cloud Questions] [answered]
Hi,

I depolyed my custom client to aws with the new endpoints (e.g. /user/$uid or /room/$scene-id) I added by editing the webpack config and register telemetry. They worked fine on my localhost. However, when I deployed to aws and went to the new endpoints on my domain, they all showed invalid url. Is there anything I should do to get the new endpoints work for my custom client on aws?

Thanks!

**Comment by robinkwilson** [comment, marked as answer]
You're receiving a "invalid url" because routing is currently managed by the internal server reticulum. Currently there's no way to update reticulum for your Hubs Cloud other than hosting a habitat package yourself. We have a feature request so webpack/the front-end manages new pages. 

You can follow this feature request here: https://github.com/mozilla/hubs-cloud/issues/106

**Reply by ylkuo** [reply]
Got it. Thank you!

So, for now, I may just not adding now pages but switch contents based on the query strings? Or, any other suggestion if I would like to add new pages?

**Reply by robinkwilson** [reply]
Yes use query strings! I've used them in my custom clients for switching content. Good thought!

**Comment by jsdude0220** [comment]
Hello,
I have also got the Invalid URL issue.
I have made the custom client page and added to the webpack.config.js, it works well in localhost:8080, but when i deploy to the hubs, it returns invalid url.
Is there a good way to add the custom client page without invalid url issue?

**Reply by robinkwilson** [reply]
Query strings could work. Thumbs up this issue to up the priority! https://github.com/mozilla/hubs-cloud/issues/106

**Reply by jsdude0220** [reply]
Thanks for your response.
what you mean the query string?
please let me know if you are available. 😊

**Reply by jsdude0220** [reply]
you mean query string parser ?
